File: src/core/leetcode/C#/MinSubArrayLen.cs
namespace NEXT.LeetCode;

// 209 长度最小的子数组
internal class MinSubArrayLen
{
    public int Main(int target, int[] nums)
    {
        int n = nums.Length, ans = int.MaxValue, start = 0, end = 0, sum = 0;

        while (end < n)
        {
            sum += nums[end];
            while (sum >= target)
            {
                ans = Math.Min(ans, end - start + 1);
                sum -= nums[start];
                start++;
            }
            end++;
        }

        return ans == int.MaxValue ? 0 : ans;
    }
}

File: src/core/leetcode/C/minSubArrayLen.c
#include <limits.h>

// 209 长度最小的子数组
int minSubArrayLen(int target, int* nums, int numsSize) {
	int minLength = INT_MAX, sum = 0, left = 0, right = 0;

	for (; right < numsSize; ++right)
	{
		sum += nums[right];
		while (sum >= target)
		{
			int subLength = right - left + 1;
			if (minLength > subLength) minLength = subLength;
			sum -= nums[left++];
		}
	}

	return minLength == INT_MAX ? 0 : minLength;
}
